src/tests/ResourceManagerTest.cpp

```diff
@@ -0,0 +1,86 @@
+#include "esp/gfx/WindowlessContext.h"
+#include "esp/scene/SceneManager.h"
+
+#include "configure.h"
+
+namespace Cr = Corrade;
+
+using esp::assets::ResourceManager;
+using esp::scene::SceneManager;
+
+const std::string dataDir = Cr::Utility::Directory::join(SCENE_DATASETS, "../");
+
+TEST(ResourceManagerTest, JoinMesh) {
+  esp::gfx::WindowlessContext::uptr context_ =
+      esp::gfx::WindowlessContext::create_unique(0);
+
+  std::shared_ptr<esp::gfx::Renderer> renderer_ = esp::gfx::Renderer::create();
+
+  // must declare these in this order due to avoid deallocation errors
+  ResourceManager resourceManager;
+  SceneManager sceneManager_;
+
+  std::string boxFile = Cr::Utility::Directory::join(
+      dataDir, "test_assets/objects/transform_box.glb");
+
+  int sceneID = sceneManager_.initSceneGraph();
+  auto& sceneGraph = sceneManager_.getSceneGraph(sceneID);
+  esp::scene::SceneNode* navSceneNode = &sceneGraph.getRootNode().createChild();
+  const esp::assets::AssetInfo info = esp::assets::AssetInfo::fromPath(boxFile);
+  resourceManager.loadScene(info, navSceneNode, nullptr);
+
+  std::shared_ptr<esp::assets::MeshData> joinedBox =
+      resourceManager.joinMesh(boxFile);
+
+  // transform_box.glb is composed of 6 identical triangulated plane meshes
+  // transformed into a cube via a transform heirarchy. Combined, the resulting
+  // mesh should have 24 vertices and 36 indices with corners at the unit corner
+  // coordinates as defined in the ground truth vectors below.
+  int numVerts = joinedBox->vbo.size();
+  int numIndices = joinedBox->ibo.size();
+
+  ASSERT_EQ(numVerts, 24);
+  ASSERT_EQ(numIndices, 36);
+
+  std::vector<Magnum::Vector3> vertGroundTruth{
+      Magnum::Vector3{-1, 1, 1},  Magnum::Vector3{-1, -1, -1},
+      Magnum::Vector3{-1, 1, -1}, Magnum::Vector3{-1, -1, 1},
+      Magnum::Vector3{1, 1, 1},   Magnum::Vector3{1, -1, -1},
+      Magnum::Vector3{1, 1, -1},  Magnum::Vector3{1, -1, 1},
+      Magnum::Vector3{1, 1, -1},  Magnum::Vector3{-1, -1, -1},
+      Magnum::Vector3{-1, 1, -1}, Magnum::Vector3{1, -1, -1},
+      Magnum::Vector3{1, 1, 1},   Magnum::Vector3{-1, 1, -1},
+      Magnum::Vector3{-1, 1, 1},  Magnum::Vector3{1, 1, -1},
+      Magnum::Vector3{1, -1, 1},  Magnum::Vector3{-1, -1, -1},
+      Magnum::Vector3{-1, -1, 1}, Magnum::Vector3{1, -1, -1},
+      Magnum::Vector3{1, 1, 1},   Magnum::Vector3{-1, -1, 1},
+      Magnum::Vector3{-1, 1, 1},  Magnum::Vector3{1, -1, 1}};
+
+  std::vector<uint32_t> indexGroundTruth{
+      0,  1,  2,  0,  3,  1,  4,  5,  6,  4,  7,  5,  8,  9,  10, 8,  11, 9,
+      12, 13, 14, 12, 15, 13, 16, 17, 18, 16, 19, 17, 20, 21, 22, 20, 23, 21};
+
+  for (size_t vix = 0; vix < joinedBox->vbo.size(); vix++) {
+    // Cr::Utility::Debug() << joinedBox->vbo[vix] << " vs " <<
+    // vertGroundTruth[vix];
+    ASSERT_EQ(vertGroundTruth[vix],
+              Magnum::Vector3(joinedBox->vbo[vix][0], joinedBox->vbo[vix][1],
+                              joinedBox->vbo[vix][2]));
```

Comment:
Could be just `Magnum::Vector3(joinedBox->vbo[vix])` if you `#include <Magnum/EigenIntegration/Integration.h>` :)